src/Cohesity.Powershell/Scripts/Vault/Get-CohesityVault.ps1

```diff
@@ -0,0 +1,45 @@
+function Get-CohesityVault {
+    <#
+        .SYNOPSIS
+        Get cohesity vault (external targets).
+        .DESCRIPTION
+        The Get-CohesityVault function is used to get all vaults.
+        .NOTES
+        Published by Cohesity
+        .LINK
+        https://cohesity.github.io/cohesity-powershell-module/#/README
+        .EXAMPLE
+        Get-CohesityVault
+        .EXAMPLE
+        Get-CohesityVault -VaultName <string>
+    #>
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory = $false)]
+        $VaultName=$null
+    )
+    Begin {
+        if (-not (Test-Path -Path "$HOME/.cohesity")) {
+            throw "Failed to authenticate. Please connect to the Cohesity Cluster using 'Connect-CohesityCluster'"
+        }
+        $session = Get-Content -Path $HOME/.cohesity | ConvertFrom-Json
+
+        $server = $session.ClusterUri
+
+        $token = $session.Accesstoken.Accesstoken
+    }
+    Process { 
+        if ($VaultName) {
+            $url = $server + '/irisservices/api/v1/public/vaults?name=' + $VaultName
+        }
+        else {
+            $url = $server + '/irisservices/api/v1/public/vaults'
+        }
```

Comment:
can we declare the url variable as common and append the name parameter if 'VaultName' exist?

Reply:
Fixed